corehq/form_processor/interfaces/case.py

```diff
@@ -0,0 +1,91 @@
+from couchdbkit import ResourceNotFound
+
+from dimagi.utils.couch.database import iter_docs
+from dimagi.utils.couch.undo import DELETED_SUFFIX
+from casexml.apps.case.models import CommCareCase
+from casexml.apps.case.dbaccessors import get_reverse_indices_for_case_id
+from casexml.apps.case.util import get_case_xform_ids
+from corehq.apps.hqcase.dbaccessors import get_case_ids_in_domain
+
+from ..utils import to_generic
+from ..exceptions import CaseNotFound
+
+
+class CaseInterface(object):
+
+    @staticmethod
+    @to_generic
+    def create_from_generic(generic_case):
```

Comment:
Side note. Instead of doing this we could always do form / case creating by submitting a form. Not sure if that's worth it but there are only 3 calls of this for cases and 3 for the forms equivalent.